### DISCUSSION TITLE
Relative Path (i.e., ../) Dependency Breaking Methods

### DISCUSSION THREAD

**Discussion by jyc-biz** [ThrowTheSwitch/Ceedling#824] [Q&A] [answered]
Hello all,

If a source file that I want to test includes relative paths, is there a way for me to break that dependency somehow?
For instance, say I want to test module.c and the project directory looks something like:
- src/
  - architecture/
    - architecture_specific.h
  - module/
    - module.c
    - module.h
  - main.c
- test/
  - support/
  - module/
    - test_module.c 
- project.yml

In module.c:
```
#include "module.h"
#include "../architecture/architecture_specific.h"
//rest of module
```
I was hoping that I could put a fake architecture_specific.h in test/support to break the lower-level dependencies, but my understanding is that since GCC will first try to look for files relative to the current file's directory, it will always find and use the real architecture_specific.h?

Therefore, the best approach I could think of is to modify the project build configurations (outside of ceedling), so that src/ is an include path, that way in module.c I can change the line to `#include "architecture/architecture_specific.h"` and add a fake "architecture/architecture_specific.h" under test/support instead.

Please let me know if there are other alternatives, because ideally I would not want to modify the build configurations or the src/ structure. Thank you in advance for your help!

**Comment by Letme** [comment, marked as answer]
Just don't use relative paths and use proper include directives to point to those directories. Mocks come first so you won't have to worry about  taking the wrong one.

So in your words: make proper build configurations. It helps when you want to refactor stuff or add more public includes.

**Reply by jyc-biz** [reply]
Sounds good, thanks so much for confirming! Just wanted to see if there might be other ways to do it.
